[Doc] Update PR checklist (#19988)
qmk/qmk_firmware: docs/pr_checklist.md

        - Microcontroller and bootloader
        - Diode Direction (if not using direct pins)
    - the following are required to be configured in `info.json` if necessary
      - Direct pin configuration
      - Backlight Configuration (where applicable)
      - Split keyboard configuration (where applicable)
      - Encoder Configuration
      - Bootmagic Configuration
      - LED Indicator Configuration
- `readme.md`
    - must follow the [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md)
    - flash command is present, and has `:flash` at end
    - valid hardware availability link (unless handwired) -- private groupbuys are okay, but one-off prototypes will be questioned. If open-source, a link to files should be provided.
    - clear instructions on how to reset the board into bootloader mode
    - a picture about the keyboard and preferably about the PCB, too
        - images are not to be placed in the `qmk_firmware` repository
        - images should be uploaded to an external image hosting service, such as [imgur](https://imgur.com/).

>         - if imgur is used, images should be resized appropriately: append "h" to the image url i.e. `https://i.imgur.com/vqgE7Ok.jpg` becomes `https://i.imgur.com/vqgE7Okh.jpg`

Reviewer: ```suggestion
        - if imgur is used, images should be resized appropriately, i.e. append "h" to the image url: [https://i.imgur.com/vqgE7Ok.jpg](https://i.imgur.com/vqgE7Ok.jpg) becomes [https://i.imgur.com/vqgE7Okh.jpg](https://i.imgur.com/vqgE7Okh.jpg)
```

>         - if imgur is used, images should be resized appropriately: append "h" to the image url i.e. `https://i.imgur.com/vqgE7Ok.jpg` becomes `https://i.imgur.com/vqgE7Okh.jpg`

Reviewer: ```suggestion
        - if imgur is used, images should be resized appropriately, i.e. append "h" to the image url: [https://i.imgur.com/vqgE7Ok.jpg](https://i.imgur.com/vqgE7Ok.jpg) becomes [https://i.imgur.com/vqgE7Okh.jpg](https://i.imgur.com/vqgE7Okh.jpg)
```

>         - image links should link directly to the image, not a "preview" (e.g. [https://imgur.com/vqgE7Ok](https://imgur.com/vqgE7Ok) should be [https://i.imgur.com/vqgE7Okh.jpg](https://i.imgur.com/vqgE7Okh.jpg) when using imgur)

Reviewer: ```suggestion
        - image links should link directly to the image, not a "preview", e.g. append the file extension to the image url when using imgur: [https://imgur.com/vqgE7Ok](https://imgur.com/vqgE7Ok) becomes [https://i.imgur.com/vqgE7Okh.jpg](https://i.imgur.com/vqgE7Okh.jpg)
```
